app/modules/transcribe/annotations/annotations.factory.js

```diff
@@ -57,8 +60,36 @@ function AnnotationsFactory(localStorageService) {
         return annotation;
     }
 
+
+    function checkVariants(annotation) {
+        //splits string by whitespace (different encodings)
+        var lemmas = annotation.text.split(/\s|&nbsp;/g);
+        var results = [];
+        var oedVars = [];
+        for (var i = 0; i < lemmas.length; ++i) {
+            //uri encode and replace all <> tags
+            var urlLemmas = encodeURIComponent(lemmas[i].replace(/<[^>]*>/g, '').toLowerCase());
+            results.push(urlLemmas);
+            $http({
+                method: 'GET',
+                url: 'https://static.zooniverse.org/www.shakespearesworld.org/variants/' + results[i] + '.txt'
+            }).then(function successCallback(response) {}, function errorCallback(response) {
+                var url = response.config.url;
+                var words = decodeURIComponent(url.substring(url.lastIndexOf("/") + 1, url.lastIndexOf(".")));
+                oedVars.push(words);
+                upsert(_.extend(annotation, {
+                    variants: oedVars
```

Comment:
Okay, this is a bit strange... assuming `oedVars` should be different for each lemma, how come you're adding the previous variants to the annotation for each new one? Shouldn't you create a new `oedVars` array for each lemma?